src/libopensc/pkcs15.c

```diff
@@ -2589,7 +2589,7 @@ sc_pkcs15_read_file(struct sc_pkcs15_card *p15card, const struct sc_path *in_pat
 		}
 		else {
 			unsigned long flags = 0;
-			r = sc_read_binary(p15card->card, offset, data, len, &flags);
+			r = sc_read_binary(p15card->card, (int)offset, data, len, &flags);
```

Comment:
Its actually `unsigned int` (not sure why I have put here the `int` -- will fix). The case is that we would have to change also the `read_binary` callbacks of all the drivers as it is directly passed which I wanted to avoid (but that can be accomplished if we want). And likely simiarly in all the API that takes index for consistency, like `sc_write_binary`, `sc_update_binary`, `sc_erase_binary` and similar API for records, which was again something I did not want to complicate this PR with (but can land in next PR or in follow-up commits if we want).